server/src/main/java/io/druid/segment/realtime/firehose/EventReceiverFirehoseFactory.java

```diff
@@ -159,6 +161,7 @@ public int getBufferSize()
     private volatile boolean closed = false;
     private final AtomicLong bytesReceived = new AtomicLong(0);
     private final AtomicLong lastBufferAddFailMsgTime = new AtomicLong(0);
+    private final ConcurrentMap<String, Long> producerSequences = new ConcurrentHashMap<>();
```

Comment:
nit: should this be some fixed size just in case clients are misbehaving and creating , say, one producerId per request.

Reply:
Do you have any suggestions on what a good size would be?

Reply:
it could be configurable and a default of 1000 should be good. I hardly see storm topologies with tranquility running on more than 300-500 nodes.

Reply:
> it could be configurable

Do you think it makes sense to have this be configurable by the end user?  As a server setting or just hard coded?

Reply:
for now, let us hardcode the size to something absurdly large ... 10,000 . we'll make it configurable if that need arises ever.

Reply:
Done